src/app/static/src/tests/integration/adr-dataset.itest.ts

```diff
@@ -204,4 +204,47 @@ describe("ADR dataset-related actions", () => {
             .toBe("anc.csv");
     }, 7000);
 
+    it("can upload files to adr", async () => {
+
+        const commit = jest.fn();
+
+        // 1. get datasets
+        await adrActions.getDatasets({commit, rootState} as any);
+
+        // 2. select a naomi dev dataset
+        const datasets = commit.mock.calls[2][0]["payload"];
+        const dataset = datasets.find((dataset: any) => dataset.organization.name === "naomi-development-team");
+
+        // 3. check can upload
+        commit.mockClear();
+        const root = {
+            ...rootState,
+            baseline: {
+                selectedDataset: {
+                    id: dataset.id,
+                    organization: {id: dataset.organization.id}
+                }
+            }
+        };
+
+        const adr = { schemas, datasets };
+
+        const uploadFilesPayload = {
+            filesToBeUploaded: [
+                {
+                    resourceType: "type1",
+                    resourceFilename: "file1",
+                    resourceId: "id1"
+                }
+            ]
+        } as UploadFilesPayload
+
+        await adrActions.uploadFilestoADR({commit, state: adr, rootState: root} as any,
+            uploadFilesPayload);
+
+        expect(commit.mock.calls[1][0]["type"]).toBe(ADRMutation.ADRUploadCompleted);
+        // expect(commit.mock.calls[1][0]["payload"]["filename"])
+        //     .toBe("anc.csv");
+    }, 7000);
+
```

Comment:
I'm not sure if I'm setting up this test correctly or what I should be receiving as a payload. Any help here would be appreciated.

Reply:
You're trying to upload a file for real here to a test dataset? You can either hardcode this I guess, or use the dataset information it's fetching at the top of the test to find the resource of an uploadable file. But you'll also need to have output data available on the server to upload (so the model run and calibrate will need to have completed). 

GIven that we're already doing upload integration tests from the backend, it mght be easier here to do a simple integration test that we're hitting the endpoint and getting an expected error.